feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,619 @@
+	handle.StreamChannelDestroy(&streamName)
+	handle.ServerDisconnect()
+}
+
+// Test Zero client with 0 electionID
+func TestZeroMaster(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
+			lowID:        *ygot.Uint64(0),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     *ygot.Uint64(1),
+			expectPass:   false,
+			expectStatus: 0,
+		},
+	}
+	var p4Clients []*p4rt_client.P4RTClient
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if err == nil && test.expectPass {
+				t.Errorf("Zero ElectionID (0,0) is allowed: %v", err)
+				removeClient(test.handle)
+			} else {
+				t.Logf("Zero ElectionID (0,0) connection failed as expected: %v", err)
+			}
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Keep track of connections for teardown
+			p4Clients = append(p4Clients, test.handle)
+		})
+	}
+}
+
+// Test Primary Reconnect
+func TestPrimaryReconnect(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		}, {
+			desc:         "Secondary connection",
+			lowID:        *ygot.Uint64(90),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  false,
+			expectRead:   true,
+			expectStatus: 5,
+		}, {
+			desc:         "Primary reconnection",
+			lowID:        *ygot.Uint64(100),
+			highID:       *ygot.Uint64(0),
+			handle:       clientConnection(t, dut),
+			deviceID:     deviceID,
+			expectPass:   true,
+			expectWrite:  true,
+			expectRead:   true,
+			expectStatus: 0,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.desc, func(t *testing.T) {
+			resp, err := streamP4RTArb(&test)
+			if err != nil && test.expectPass {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			// Validate status code
+			if resp != test.expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.expectStatus, resp)
+			}
+			// Validate the response for Read/Write for each client
+			validateRWResp(t, &test)
+			// Disconnect Primary
+			removeClient(test.handle)
+		})
+	}
+}
+
+// Test Primary/Secondary clients
+func TestPrimarySecondary(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
+		{
+			desc:         "Primary connection",
```

Comment:
Similar to the other comments below. Can we only add the variables in the table here.